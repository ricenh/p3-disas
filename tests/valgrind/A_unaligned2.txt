==1603066== Memcheck, a memory error detector
==1603066== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1603066== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1603066== Command: ../y86 -a -d inputs/unaligned2.o
==1603066== 
01 00 55 05 10 00 03 00  00 00 00 00 45 4c 46 00
Mini-ELF version 1
Entry point 0x555
There are 3 program headers, starting at offset 16 (0x10)
There is no symbol table present
There is no string table present
 Segment   Offset    Size      VirtAddr  Type      Flags
  00       0x004c    0x0009    0x0000    CODE      R X
  01       0x0055    0x001f    0x0555    CODE      R X
  02       0x0074    0x0008    0x0987    DATA      RW 
Contents of memory from 0000 to 0009:
  0000  70 55 05 00 00 00 00 00  00
Contents of memory from 0555 to 0574:
  0550                 30 f0 4d  01 00 00 00 00 00 00 30
  0560  f3 de 00 00 00 00 00 00  00 30 f1 6f 00 00 00 00
  0570  00 00 00 00
Contents of memory from 0987 to 098f:
  0980                       dd  cc bb aa 00 00 00 00
Disassembly of executable contents:
  0x000:                               | .pos 0x000 code
  0x000: 70 55 05 00 00 00 00 00 00    |   jmp 0x555

  0x555:                               | .pos 0x555 code
  0x555:                               | _start:
  0x555: 30 f0 4d 01 00 00 00 00 00 00 |   irmovq 0x14d, %rax
  0x55f: 30 f3 de 00 00 00 00 00 00 00 |   irmovq 0xde, %rbx
  0x569: 30 f1 6f 00 00 00 00 00 00 00 |   irmovq 0x6f, %rcx
  0x573: 00                            |   halt

==1603066== 
==1603066== HEAP SUMMARY:
==1603066==     in use at exit: 0 bytes in 0 blocks
==1603066==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1603066== 
==1603066== All heap blocks were freed -- no leaks are possible
==1603066== 
==1603066== For lists of detected and suppressed errors, rerun with: -s
==1603066== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
