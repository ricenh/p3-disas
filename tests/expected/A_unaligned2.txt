01 00 55 05 10 00 03 00  00 00 00 00 45 4c 46 00
Mini-ELF version 1
Entry point 0x555
There are 3 program headers, starting at offset 16 (0x10)
There is no symbol table present
There is no string table present
 Segment   Offset    Size      VirtAddr  Type      Flags
  00       0x004c    0x0009    0x0000    CODE      R X
  01       0x0055    0x001f    0x0555    CODE      R X
  02       0x0074    0x0008    0x0987    DATA      RW 
Contents of memory from 0000 to 0009:
  0000  70 55 05 00 00 00 00 00  00
Contents of memory from 0555 to 0574:
  0550                 30 f0 4d  01 00 00 00 00 00 00 30
  0560  f3 de 00 00 00 00 00 00  00 30 f1 6f 00 00 00 00
  0570  00 00 00 00
Contents of memory from 0987 to 098f:
  0980                       dd  cc bb aa 00 00 00 00
Disassembly of executable contents:
  0x000:                               | .pos 0x000 code
  0x000: 70 55 05 00 00 00 00 00 00    |   jmp 0x555

  0x555:                               | .pos 0x555 code
  0x555:                               | _start:
  0x555: 30 f0 4d 01 00 00 00 00 00 00 |   irmovq 0x14d, %rax
  0x55f: 30 f3 de 00 00 00 00 00 00 00 |   irmovq 0xde, %rbx
  0x569: 30 f1 6f 00 00 00 00 00 00 00 |   irmovq 0x6f, %rcx
  0x573: 00                            |   halt

